test/integration/sonobuoy_integration_test.go

```diff
@@ -32,6 +34,42 @@ func findSonobuoyCLI() (string, error) {
 	return sonobuoyPath, nil
 }
 
+func cleanup(t *testing.T, namespace string) {
+	var stdout, stderr bytes.Buffer
+	deleteCommand := exec.Command(sonobuoy, "delete", "--wait", "-n", namespace)
+	deleteCommand.Stdout = &stdout
+	deleteCommand.Stderr = &stderr
+
+	t.Logf("Cleaning up. Running %q\n", deleteCommand.String())
+	if err := deleteCommand.Run(); err != nil {
```

Comment:
I think as part of expanding this suite we need to make a handful of helper functions that handle this sort of thing:
 - run, log output if error & error a test
 - run, return output

It feels like you're having to do too much right now each time you want to execute some sonobuoy action.

I'd rather have a test that looks like
```
func TestFoo(...){
  cmd:="run --plugin foo1 --plugin foo2"
  expectInResults := []string{"s1","s2"}
  output := mustExecuteCommandAndGetResults(t, cmd)
  for _,s:=range expectInResults{
    if !strings.Contains(s1, output) {...}
  }
}
```

Once we centralize more of the boilerplate of dealing namespaces, buffers, etc into a few of these helpers we can have tests that look almost like unit tests (e.g. table driven tests to just check for strings/files/etc).

Reply:
Yes, I definitely wanted to do that. I had started to do that, then got a bit apprehensive because I was basing the design on one test. I can add some of this now though and refactor later.

Reply:
Valid concern; we can follow up with a handful of useful tests and the helpers they need next even in the same PR.

You're right not to want to prematurely make 100 hepler funcs which we dont really need.